### PR TITLE
feat: 2020 Add a performance test for iOS in Bitrise

### PR DESCRIPTION
**Pull request by MarioAslau**
## **Description**

iOS cold app start slightly regressed recently. We want to prevent that from happening again by adding a performance test for iOS in Bitrise, similar to the one we've setup for Android.

This PR includes:
- Appium Upgrade to v2.0
- iOS Browserstack configuration update
- Fixing Flaky Android Regression Test

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: wdio/config/ios.config.browserstack.js [hunk 1]
```diff
@@ -9,27 +9,78 @@ const browserstack = require('browserstack-local');
...
     settings: {
       snapshotMaxDepth: 100,
     },
+  }
+];
+
+// Define capabilities for app upgrade tests
+const upgradeCapabilities = [
+  {
+    platformName: 'Android',
```
Comment (cortisiko) [inline wdio/config/ios.config.browserstack.js:34]: update the platform name

File: wdio/config/ios.config.browserstack.js [hunk 2]
```diff
@@ -9,27 +9,78 @@ const browserstack = require('browserstack-local');
 
 config.user = process.env.BROWSERSTACK_USERNAME;
 config.key = process.env.BROWSERSTACK_ACCESS_KEY;
-config.capabilities = [
+
+const defaultCapabilities = [
   {
-    platformName: 'iOS',
+    platformName: 'Android',
```
Comment (cortisiko) [inline wdio/config/ios.config.browserstack.js:15]: update the platform name

File: wdio/config/ios.config.browserstack.js [hunk 3]
```diff
@@ -9,27 +9,78 @@ const browserstack = require('browserstack-local');
...
     noReset: false,
     fullReset: false,
     maxInstances: 1,
-    build: 'iOS QA E2E Tests',
-    automationName: 'XCUITest',
-    deviceName: 'iPhone 13 Pro',
-    platformVersion: '16',
-    app: process.env.BROWSERSTACK_APP_URL, // TODO: Add package ID when upload to BrowserStack
+    build: 'iOS App Launch Times Tests',
+    device: process.env.BROWSERSTACK_DEVICE || 'iPhone 13 Pro',
```
Comment (cortisiko) [inline wdio/config/ios.config.browserstack.js:20]: readd `automationName: 'XCUITest',`

**Review by cortisiko — COMMENTED** [review]
update the performance feature files with the iosApp tag.

File: wdio/config/ios.config.browserstack.js [hunk 4]
```diff
@@ -9,27 +9,78 @@ const browserstack = require('browserstack-local');
...
+    platformName: 'Android',
     noReset: false,
     fullReset: false,
     maxInstances: 1,
-    build: 'iOS QA E2E Tests',
-    automationName: 'XCUITest',
-    deviceName: 'iPhone 13 Pro',
-    platformVersion: '16',
-    app: process.env.BROWSERSTACK_APP_URL, // TODO: Add package ID when upload to BrowserStack
+    build: 'iOS App Launch Times Tests',
```
Comment (cortisiko) [inline wdio/config/ios.config.browserstack.js:19]: readd     `automationName: 'XCUITest',`

File: wdio/config/ios.config.browserstack.js [hunk 5]
```diff
@@ -9,27 +9,78 @@ const browserstack = require('browserstack-local');
...
+];
+
+// Define capabilities for app upgrade tests
+const upgradeCapabilities = [
+  {
+    platformName: 'Android',
+    noReset: false,
+    fullReset: false,
+    maxInstances: 1,
+    build: 'iOS App Upgrade Tests',
```
Comment (cortisiko) [inline wdio/config/ios.config.browserstack.js:38]: add  `automationName: 'XCUITest',`

**Comment by Cal-L** [comment]
@MarioAslau I believe most of the package updates are dev dependencies

**Comment by MarioAslau** [comment]
Current Bitrise Performance Pipeline Being Run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/25df7847-0479-4c62-aa4d-1f643e611487?tab=workflows

**Review by cortisiko — APPROVED** [review]
🌮 🌮

**Pull request merged**